File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.py
class Solution:
    def answerQueries(self, nums: List[int], queries: List[int]) -> List[int]:
        def bs(a,val):
            l,r=0,len(a)-1
            while l<=r:
                mid=(l+r)//2
                if a[mid]<=val:
                    l=mid+1
                else:
                    r=mid-1
            return l
            
        nums.sort()
        for i in range(1,len(nums)):
            nums[i]+=nums[i-1]
        ans=[]
        for i in queries:
            ans.append(bs(nums,i))
        return ans
        
        

File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.cpp
class Solution {
public:
    vector<int> answerQueries(vector<int>& nums, vector<int>& q) 
    {
        sort(nums.begin(),nums.end());
        for(int i=1,n=nums.size();i<n;i++)
        {
            nums[i]+=nums[i-1];
        }
        vector<int>ans;
        for(auto it:q)
        {
            int index=upper_bound(nums.begin(),nums.end(),it)-nums.begin();
            ans.push_back(index);
        }
        return ans;
    }
};